docs/guides/kotlin.md

```diff
@@ -0,0 +1,575 @@
+
+## 1.1 Installing Java and Gradle
+
+Please refer to [the Oracle's guide for installing JDK](https://www.oracle.com/technetwork/java/javase/downloads/index.html).
+
+Verify that you have installed Java successully:
+
+```sh
+$ java -version
+java version "1.8.0_162"
+Java(TM) SE Runtime Environment (build 1.8.0_162-b12)
+Java HotSpot(TM) 64-Bit Server VM (build 25.162-b12, mixed mode)
+```
+
+You can choose any version of Java higher or equal to 8.
+In my case it is Java SE Development Kit 8.
+
+Make sure you have `$JAVA_HOME` environment variable set:
+
+```sh
+$ echo $JAVA_HOME
+/Library/Java/JavaVirtualMachines/jdk1.8.0_162.jdk/Contents/Home
+```
+
+For Gradle installation please refer to [their official guide](https://gradle.org/install/).
+
+## 1.2 Creating a new Kotlin project
+
+We'll start by creating a new Gradle project.
+
+```sh
+$ export KVSTORE_HOME=~/kvstore
+$ mkdir $KVSTORE_HOME
+$ cd $KVSTORE_HOME
+```
+
+Inside the example directory run:
+```sh
+gradle init --dsl groovy --package io.example --project-name example --type kotlin-application
+```
+That Gradle command will create project structure for you:
+```sh
+$ tree
+.
+|-- build.gradle
+|-- gradle
+|   `-- wrapper
+|       |-- gradle-wrapper.jar
+|       `-- gradle-wrapper.properties
+|-- gradlew
+|-- gradlew.bat
+|-- settings.gradle
+`-- src
+    |-- main
+    |   |-- kotlin
+    |   |   `-- io
+    |   |       `-- example
+    |   |           `-- App.kt
+    |   `-- resources
+    `-- test
+        |-- kotlin
+        |   `-- io
+        |       `-- example
+        |           `-- AppTest.kt
+        `-- resources
+```
+
+When run, this should print "Hello world." to the standard output.
+
+```sh
+$ ./gradlew run
+> Task :run
+Hello world.
+```
+
+## 1.3 Writing a Tendermint Core application
+
+Tendermint Core communicates with the application through the Application
+BlockChain Interface (ABCI). All message types are defined in the [protobuf
+file](https://github.com/tendermint/tendermint/blob/develop/abci/types/types.proto).
+This allows Tendermint Core to run applications written in any programming
+language.
+
+### 1.3.1 Compile .proto files
+
+Add folowing to the top of `build.gradle`:
+```groovy
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.8'
+    }
+}
+```
+
+Enable protobuf plugin in `plugins` section of `build.gradle`:
+```groovy
+plugins {
+    id 'com.google.protobuf' version '0.8.8'
+}
+```
+
+Add following to `build.gradle`:
+```groovy
+protobuf {
+    protoc {
+        artifact = "com.google.protobuf:protoc:3.7.1"
+    }
+    plugins {
+        grpc {
+            artifact = 'io.grpc:protoc-gen-grpc-java:1.22.1'
+        }
+    }
+    generateProtoTasks {
+        all()*.plugins {
+            grpc {}
+        }
+    }
+}
+```
+
+Now your project is ready to compile `*.proto` files.
+
+
+Copy necessary .proto files to your project:
+```sh
+mkdir -p \
```

Comment:
If I understood you correctly I've already done it [here](https://github.com/climber73/tendermint-abci-grpc-kotlin). Or do you mean something else?

Reply:
Awesome! My question is then why does a user need to copy proto files again? Can't it just reuse them from your plugin?

Reply:
When this guide was being written, I assumed that the user would perform all actions from scratch step by step. Therefore, I thought that the only goal of the guide was to give an understanding of how to develop an ABCI app. Do you think it's better to provide user with project to start from? It might be good idea. Then I have to rework a bit that sample project. And this guide too.

Reply:
It may be fine as it is. Libraries like https://github.com/jTendermint/jabci sometime get abandoned. So it might be better to avoid this "extra dependency".

Just for the info: the idea behind many plugins is to provide ABCI servers in target languages (C++, Java, Rust, ...) and compiled protobuf ABCI classes (to lessen the burden of app developers. i.e. removing the need to execute `gradle generateProto`).

> Do you think it's better to provide user with project to start from?

Maybe we can do it separately. What do you think?

Reply:
If I were 100% sure that the .proto files would never change, I would commit the generated ABCI classes and forget about them. But who knows, maybe ABCI messages will be changed some day. And nobody will know how to update ABCI classes :) 

> Maybe we can do it separately. What do you think?

Sure I'll try.

By the way, is Java guide still needed (in the form in which this guide is made)?

Reply:
> By the way, is Java guide still needed (in the form in which this guide is made)?

Of course!